### PR TITLE
Add deploy manifest handler for GH Copilot

### PR DESCRIPTION
**Pull request by kejatura-dev**
PR introduces deploy manifest to an AKS cluster handler for GH Copilot:

image

PR adds:
- New plugin will be invoked from @azure participant

### CODE REVIEW

File: src/plugins/shared/pluginResponses.ts [hunk 1]
```diff
@@ -11,4 +11,13 @@ export const getAKSClusterPluginResponse = (): GitHubCopilotForAzureChatPluginRe
         commandID: "aks.aksCreateClusterFromCopilot",
         buttonLabel: "Get started"
     }
+}
+
+//Deploy manifest to AKS cluster plugin
```
Comment (kejatura-dev) [inline src/plugins/shared/pluginResponses.ts:16]: Copying comment over from https://github.com/Azure/vscode-aks-tools/pull/966#discussion_r1790934232: 

deploying an application to AKS requires few more steps, I agree this is all for POC purposes

App image should be deployed beforehand to container registry, can you include those as well ?
if it is ACR (Azure Container Registry), acrpull role assignment should be added for the user to deploy to AKS successfully.

File: src/commands/aksDeployManifest/aksDeployManifest.ts [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    if(!checkGitHubCopilotExtension) {
+        handleExtensionDoesNotExist(GITHUBCOPILOT_FOR_AZURE_VSCODE_ID);
+        return;
+    }
+
+    // Select manifest
+    const manifest = await getManifestFile();
+
+    if(manifest === undefined) { 
+        return;
```
Comment (hsubramanianaks) [inline src/commands/aksDeployManifest/aksDeployManifest.ts:29]: show a useful message to user that manifest files doesn't exist?

Comment (kejatura-dev) [reply src/commands/aksDeployManifest/aksDeployManifest.ts:29]: This is shown in the getManifestFile function. This is what it looks like: 
image

Comment (Tatsinnit) [reply src/commands/aksDeployManifest/aksDeployManifest.ts:29]: 💡 Just a side note: Please make sure the message is informative enough for user to know what action is intended, might be get some messaging which fills that gap.

File: src/commands/aksDeployManifest/aksDeployManifest.ts [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        placeHolder: "Select manifest to deploy",
+    });
+
+    if (!fileSelected) {
+        vscode.window.showWarningMessage("Manifest file not selected");
+        return;
+    }
+
+    return fileSelected.filePath;
+}
```
Comment (Tatsinnit) [inline src/commands/aksDeployManifest/aksDeployManifest.ts:106]: 💡 Suggestion/observation: for little rewrite of function above for 4 reasons:

- `Error Handling`: Added a try/catch block to handle any errors during the file search process.
- `Code Simplification`: The mapping of files to items was simplified using the .map function.
- `Sorting`: The sorting now uses a locale-based comparison (localeCompare), which ensures a consistent sort order.
- `Early Exits`: Simplified early return points in the function when no files are found or when no file is selected.

```*.ts
type ManifestQuickPickItem = vscode.QuickPickItem & { filePath: string };

async function getManifestFile(): Promise<string | undefined> {
    try {
        // Find all YAML files in the workspace, excluding node_modules
        const files = await vscode.workspace.findFiles("**/*.yaml", "**/node_modules/**");
        
        // If no files are found, show a warning and exit
        if (files.length === 0) {
            vscode.window.showWarningMessage("No manifest files found in the workspace.");
            return;
        }

        // Map the found files to QuickPick items
        const items: ManifestQuickPickItem[] = files.map(fileUri => ({
            label: path.basename(fileUri.fsPath),
            description: fileUri.fsPath,
            filePath: fileUri.fsPath
        }));

        // Show the QuickPick to the user to select a file
        const fileSelected = await vscode.window.showQuickPick(items.sort((a, b) => a.label.localeCompare(b.label)), {
            title: "Select YAML",
            placeHolder: "Select manifest to deploy"
        });

        // If no file was selected, show a warning and exit
        if (!fileSelected) {
            vscode.window.showWarningMessage("Manifest file not selected");
            return;
        }

        return fileSelected.filePath;
    } catch (error) {
        vscode.window.showErrorMessage(`Error finding manifest files: ${error.message}`);
    }
}

```

File: src/commands/aksDeployManifest/aksDeployManifest.ts [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+async function confirmDeployment(clusterName: string): Promise<boolean> {
```
Comment (Tatsinnit) [inline src/commands/aksDeployManifest/aksDeployManifest.ts:108]: 💡Comment only for fyi and latter handling:  Lets treat this one latter but a suggestion is, if this pattern is used various places we can enum this `"yes"` and `"no"` but leaving this as comment only.

File: src/commands/aksDeployManifest/aksDeployManifest.ts [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+    const confirmed = await vscode.window.showQuickPick(["Yes", "No"], {
```
Comment (Tatsinnit) [inline src/commands/aksDeployManifest/aksDeployManifest.ts:109]: 💡 Nit: purely for refactor sake latter, aka lets handle latter if you want: at this point it is not `confirmed` I think it still a selection, but its file its a nomenclature for code readability. Thanks heaps.

File: src/commands/aksDeployManifest/aksDeployManifest.ts [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+
+    if (!kubectl.available) {
+        vscode.window.showWarningMessage(`Kubectl is unavailable.`);
+        return { succeeded: false, error: "Kubectl is unavailable." };
+    }
+
+    const kubeConfigFile = await createTempFile(cluster.kubeConfigYAML, "yaml");
+
+    const result = await longRunning(`Deployment of application to cluster: ${cluster.clusterName} in progress`, async () => {
+        return await invokeKubectlCommand(kubectl, kubeConfigFile.filePath, `apply -f ${manifestPath}`);
```
Comment (Tatsinnit) [inline src/commands/aksDeployManifest/aksDeployManifest.ts:136]: 💡 Note please: this long running already awaits for promise to be returned:

So, you can remove the return here

```.ts
    // Execute the deployment command
    const result = await longRunning(
        `Deploying application to cluster: ${cluster.clusterName} in progress...`,
        async () => invokeKubectlCommand(kubectl, kubeConfigFile.filePath, `apply -f ${manifestPath}`)
    );
```

Here is bit of a reconstruction of this function with comments, if this help thanks.

```*.ts
type DeployApplicationToClusterResult = { url: string };

async function deployApplicationToCluster(
    params: DeployApplicationToClusterOptions
): Promise<Errorable<DeployApplicationToClusterResult>> {
    const { cluster, manifestPath } = params;

    // Ensure kubectl is available
    const kubectl = await k8s.extension.kubectl.v1;
    if (!kubectl.available) {
        vscode.window.showWarningMessage("Kubectl is unavailable.");
        return { succeeded: false, error: "Kubectl is unavailable." };
    }

    // Create a temporary kubeconfig file
    const kubeConfigFile = await createTempFile(cluster.kubeConfigYAML, "yaml");

    // Execute the deployment command
    const result = await longRunning(
        `Deploying application to cluster: ${cluster.clusterName} in progress...`,
        async () => invokeKubectlCommand(kubectl, kubeConfigFile.filePath, `apply -f ${manifestPath}`)
    );

    // Check for errors during the kubectl command execution
    if (failed(result)) {
        return { succeeded: false, error: result.error };
    }

    // Generate and return the resource URL
    const resourceUrl = getPortalResourceUrl(getEnvironment(), cluster.clusterId);
    return { succeeded: true, result: { url: resourceUrl } };
}
```

**Review by Tatsinnit — APPROVED** [review]
❤️ Looks good to me, and thank you so much for kind pings, I have recommended few things as improvements and please take a look. It will be worth to get decent testing and I am happy to lend any hands if you want.

File: src/commands/aksDeployManifest/aksDeployManifest.ts (hunk 5 above)
Comment (kejatura-dev) [reply src/commands/aksDeployManifest/aksDeployManifest.ts]: Thanks, renamed to "selection" to make it more clear

File: src/commands/aksDeployManifest/aksDeployManifest.ts (hunk 4 above)
Comment (kejatura-dev) [reply src/commands/aksDeployManifest/aksDeployManifest.ts:123]: Added enum

File: src/commands/aksDeployManifest/aksDeployManifest.ts (hunk 3 above)
Comment (kejatura-dev) [reply src/commands/aksDeployManifest/aksDeployManifest.ts:121]: Thanks for this! Included in recent push

File: src/commands/aksDeployManifest/aksDeployManifest.ts (hunk 2 above)
Comment (kejatura-dev) [reply src/commands/aksDeployManifest/aksDeployManifest.ts:29]: Added a little more description for clarity on action

**Review by Tatsinnit — DISMISSED** [review]

**Review by hsubramanianaks — DISMISSED** [review]
Thanks @kejatura-dev.

**Review by hsubramanianaks — APPROVED** [review]

**Review by Tatsinnit — APPROVED** [review]

**Pull request merged**